src/glmfit.jl

```diff
@@ -543,17 +544,15 @@ function fit(::Type{M},
     end
 
     # For backward compatibility accept wts as AbstractArray and coerce them to FrequencyWeights
-    _wts = if wts isa AbstractWeights
-        wts
-    elseif wts isa AbstractVector
-        Base.depwarn("Passing weights as vector is deprecated in favor of explicitly using " *
-                     "`AnalyticWeights`, `ProbabilityWeights`, or `FrequencyWeights`. Proceeding " *
-                     "by coercing `wts` to `FrequencyWeights`",
+    _wts = convert_weights(wts)
+    if isempty(_wts)
```

Comment:
This introduces a type instability even for users who don't use the deprecated form. Better add a check on the type to avoid this:
```suggestion
    if !wts isa AbstractWeights && isempty(_wts)
```

BTW, wouldn't it be easier to do this inside `convert_weights` (passing it the expected length)?

Reply:
Done